### ISSUE TITLE
allow to pass value as props

### ISSUE THREAD

**Issue by fdecampredon** [jquery-support/react-globalize#34]
```javascript
<FormatDate options={{date: 'short'}} >{date}</FormatDate>
```
can feel a bit awkward could this module also support  this api ? 
```javascript 
<FormatDate value={date} options={{date: 'short'}} />
```

**Comment by rxaviers** [comment]
Hi,

Currently, whatever is passed in `value` gets overwritten by `children` (code below). We could change the below for not overwritten `args[0]` if `children` is falsy. so we enable both usages in your description.

```js
// src/generator.js:
this.args[0] = this.props.children;
```

Would you like to submit a pull request with such change?

**Comment by fdecampredon** [comment]
I'll do so !

**Comment by kborchers** [comment]
Hi @fdecampredon. Are you still interested in working on this?

**Comment by cowboy** [comment]
I had the same question. I'd love to see this implemented. Basically, use `this.props.value` if set, otherwise use `this.props.children`.

**Comment by gnarf** [comment]
I actually think that you should only allow `value=` prop and deprecate the `children` format.

**Comment by cowboy** [comment]
> I actually think that you should only allow value= prop and deprecate the children format.

:+1:  to that

**Comment by rxaviers** [comment]
@gnarf I see you have a preference, but the motivation isn't clear to me. Using a value attribute instead of children might be convenient in the given example (I agree), but using the opposite (children instead of a value attribute) might be convenient in other cases like:

```jsx
<FormatMessage>You’re receiving notifications because you commented</FormatMessage>
```

I hold my preference for "use `this.props.value` if set, otherwise use `this.props.children`". I don't see any problem with allowing both options this way.

PRs are still very welcome. Thanks

**Comment by kborchers** [comment]
I agree with @rxaviers here. I am working on adding some initial tests to the repo (PR almost ready) and then I will add this "use `this.props.value` if set, otherwise use `this.props.children`" functionality along with additional tests
